pkg/common/cns-lib/volume/manager.go

```diff
@@ -951,3 +956,54 @@ func (m *defaultManager) ConfigureVolumeACLs(ctx context.Context, spec cnstypes.
 	}
 	return err
 }
+
+// RegisterDisk registers a virtual disk as a First Class Disk.
+// This method helps in registering VCP volumes as FCD using vslm endpoint
+func (m *defaultManager) RegisterDisk(ctx context.Context, path string, name string) (string, error) {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return "", err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectVslm(ctx)
+	if err != nil {
+		log.Errorf("ConnectVslm failed with err: %+v", err)
+		return "", err
+	}
+	globalObjectManager := vslm.NewGlobalObjectManager(m.virtualCenter.VslmClient)
+	vStorageObject, err := globalObjectManager.RegisterDisk(ctx, path, name)
+	if err != nil {
+		alreadyExists, objectID := cnsvsphere.IsAlreadyExists(err)
+		if alreadyExists {
+			log.Infof("vStorageObject: %q, already exists and registered as First Class Disk. Returning success for RegisterDisk operation", objectID)
+			return objectID, nil
+		}
+		log.Errorf("failed to register virtual disk %q as first class disk with err: %v", path, err)
+		return "", err
+	}
+	return vStorageObject.Config.Id.Id, nil
+}
+
+// RetrieveVStorageObject helps in retreiving virtual disk information for a given volume id
+func (m *defaultManager) RetrieveVStorageObject(ctx context.Context, volumeID string) (*vim25types.VStorageObject, error) {
```

Comment:
shouldn't these go into govmomi repo?

Reply:
No, this is just a wrapper to return VStorageObject.